File: django_ts_gems/products/models/product.py
from django.db import models


class Product(models.Model):
    first_image = models.URLField()

    second_image = models.URLField()

    category = models.ForeignKey(
        to='products.Category',
        on_delete=models.CASCADE,
    )

    collection = models.ForeignKey(
        to='products.Collection',
        on_delete=models.CASCADE,
    )

    material = models.ForeignKey(
        to='products.Material',
        on_delete=models.CASCADE,
    )

    reference = models.ForeignKey(
        to='products.Reference',
        on_delete=models.CASCADE,
    )

    stones_colors = models.ManyToManyField(
        to='products.StonesColors',
    )

    def __str__(self):
        return f'{self.collection} {self.reference} {self.category}'

File: client/src/app/types/product.ts
export interface PaginatedProducts {
    count: number;
    next: string | null;
    previous: string | null;
    results: Product[];
}

export interface Product {
    id: number;
    first_image: string;
    second_image: string;
    category: number;
    collection: number;
    material: number;
    reference: number;
    stones_colors: number[];
}
